include/openenclave/edger8r/enclave.h

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * @file enclave.h
+ *
+ * This file defines the internal functions and data-structures used by
+ * oeedger8r generated code on the enclave side.
+ * These internals are subject to change without notice.
+ *
+ */
+#ifndef _OE_EDGER8R_ENCLAVE_H
+#define _OE_EDGER8R_ENCLAVE_H
+
+#ifdef _OE_EDGER8R_HOST_H
+#error \
+    "edger8r/enclave.h and edger8r/host.h must not be included in the same compilation unit."
+#endif
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * The type of a function in ecall function table
+ */
+typedef void (*oe_ecall_func_t)(void*);
+
+/**
+ * Perform a high-level host function call (OCALL).
```

Comment:
we may need to tell doxygen not to catalog these APIs as they are kind of internal

Reply:
Good point. In our case, the documented files are whitelisted and luckily the new files won't get doxygenated.